### ISSUE TITLE
Check if restframe model can cover all observed wavelengths

### ISSUE THREAD

**Issue by pmelchior** [pmelchior/spender#34]
One can easily construct requests to generate an observed spectrum with a redshift that exposes wavelengths shorter than the blue end of `wave_rest`. More generally, it's not guaranteed that the resampling method in the generator limits itself to the wavelength covered by the restframe model.

There should be at least a warning, but ideally the generate creates 0 for both spectrum and weights generated this way. The former can be done in `Decoder.transform`, but it doesn't deal with `w`. The other alternative would be to deal with it in `BaseAutoencoder.loss`, so that there can be garbage in the generated spectrum, but the loss would not care. Something like 

```
valid = instrument.wave_obs[None, :] > self.wave_rest.min() * (1 + z[:,None])
valid &= instrument.wave_obs[None, :] < self.wave_rest.max() * (1 + z[:,None]) 
torch.sum(0.5 * valid * w * (x - spectrum_observed).pow(2), dim=1) / x.shape[1]
```
